### PR TITLE
Dashboard: Add ability to stop title/description generation

### LINKED ISSUE #76984: Auto-generate: Implement "stop" generation functionality
Investigate adding the ability to "stop" generation mid-request.

This is a feature that openAI supports in ChatGPT and is something we would like to have in place before ObsCon

(spinner / Stop generating text)

https://github.com/grafana/grafana/assets/22381771/d0269e8e-b521-4432-867f-e79a01b9eefc

### PR DESCRIPTION
**Pull request by adela-almasan**
Stop generation mid-request functionality.

https://github.com/grafana/grafana/assets/88068998/c6bc4158-af4c-44fa-8c49-02bf67aa2ca9

Fixes #76984

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: public/app/features/dashboard/components/GenAI/GenAIButton.tsx [hunk 1]
```diff
@@ -197,4 +202,7 @@ const getStyles = (theme: GrafanaTheme2) => ({
   wrapper: css({
     display: 'flex',
   }),
+  spinner: css({
+    color: '#6E9FFF',
```
Comment (nmarrs) [inline public/app/features/dashboard/components/GenAI/GenAIButton.tsx:206]: does this color come from theme? (sometimes figma will share this info / if it comes from a theme preset)

File: public/app/features/dashboard/components/GenAI/GenAIHistory.tsx [hunk 2]
```diff
@@ -80,7 +79,12 @@ export const GenAIHistory = ({
   };
 
   const onApply = () => {
-    onApplySuggestion(history[currentIndex - 1]);
+    if (isStreamGenerating) {
+      setShouldStop(true);
+      updateHistory(sanitizeReply(reply));
```
Comment (nmarrs) [inline public/app/features/dashboard/components/GenAI/GenAIHistory.tsx:84]: maybe we should only update history if the reply is not `''` 🤔 ran into this edge case while testing locally

**Review by nmarrs — COMMENTED** [review]
Overall works really well! Just a few nits 💯 great job tackling / fixing up the tests ;) didn't seem too bad in the end 😅

File: public/app/features/dashboard/components/GenAI/GenAIButton.tsx (hunk 1 above)
Comment (adela-almasan) [reply public/app/features/dashboard/components/GenAI/GenAIButton.tsx:206]: oops, should've done a better job at cleaning it up, made the other PR and forgot to return to this one.. 😬 updated.

**Review by nmarrs — APPROVED** [review]
🦃

**Pull request merged**